SYCL/ESIMD/api/functional/functions/functions_select_rvalue.hpp

```diff
@@ -0,0 +1,204 @@
+                  "size should be less than char max value.");
+
+    shared_allocator<DataT> allocator(queue);
+    shared_vector<DataT> result(NumElems, allocator);
+    shared_vector<DataT> initial_ref_data(NumElems, allocator);
+    shared_vector<DataT> ref_data_for_fill(NumElems, allocator);
+
+    shared_element<bool> is_correct_type(queue, true);
+
+    for (size_t i = 0; i < NumElems; ++i) {
+      initial_ref_data[i] = i + 1;
+    }
+    // We should have different values in the first reference data and in the
+    // second reference data.
+    for (size_t i = 0; i < NumElems; ++i) {
+      ref_data_for_fill[i] =
+          initial_ref_data[i] + value_for_increase_ref_data_for_fill;
+    }
+
+    queue.submit([&](sycl::handler &cgh) {
+      DataT *init_ref_ptr = initial_ref_data.data();
+      DataT *ref_data_for_fill_ptr = ref_data_for_fill.data();
+      DataT *const out_ptr = result.data();
+      auto is_correct_type_ptr = is_correct_type.data();
+
+      cgh.single_task<
+          Kernel<DataT, NumElems, TestCaseT, NumSelectedElemsT, StrideT>>(
+          [=]() SYCL_ESIMD_KERNEL {
+            *is_correct_type_ptr =
+                TestCaseT::template call_operator<DataT, NumElems,
+                                                  NumSelectedElems, Stride>(
+                    init_ref_ptr, ref_data_for_fill_ptr, out_ptr, offset);
+          });
+    });
+    queue.wait_and_throw();
+
+    // Verify that selected values has been changed to values from referece data
+    // for fill.
+    for (size_t i = offset; i < Stride * NumSelectedElems + offset;
+         i += Stride) {
+      DataT expected = ref_data_for_fill[(i - offset) / Stride];
+      DataT retrieved = result[i];
+      if (expected != retrieved) {
+        passed = fail_test(i, expected, retrieved, data_type);
+      }
+    }
+    // Verify that values, that do not was selected has initial values.
+    for (size_t i = 0; i < NumElems; ++i) {
+      if (i % Stride != 0) {
+        DataT expected = initial_ref_data[i];
+        DataT retrieved = result[i];
+        if (expected != retrieved) {
+          passed = fail_test(i, expected, retrieved, data_type);
+        }
+      }
+    }
+
+    if (!is_correct_type.value()) {
+      passed = false;
+      log::note("Test failed due to type of the object that returns " +
+                TestCaseT::get_description() +
+                " operator is not equal to the expected one for simd<" +
+                data_type + ", " + std::to_string(NumElems) + ">.");
+    }
+
+    return passed;
+  }
+
+private:
+  bool fail_test(size_t i, DataT expected, DataT retrieved,
+                 const std::string &data_type) {
+    const auto description = TestDescription<DataT, NumElems>(
+        i, retrieved, expected, NumSelectedElems, Stride, data_type);
+    log::fail(description);
+
+    return false;
+  }
+};
+
+template <tested_types TestedTypesT>
+bool run_test_with_chosen_data_types(sycl::queue &queue) {
```

Comment:
Just as a suggestion:
```suggestion
bool run_test_for_types(sycl::queue &queue) {
```

Reply:
The function has been renamed [here](https://github.com/intel/llvm-test-suite/pull/875/commits/9519eccf663e28b1760cf5bf869ba75ea5fa275d#diff-ee52ba82babf88cc257b75cc71c61eca3df7aa19a2b6ce7b5ed19349642a56f6R183)